src/dashboard/apps/core/models.py

```diff
@@ -159,10 +166,47 @@ def get_validated_consents(self) -> QuerySet:
         return self.get_consents(obj=Consent.validated_objects).order_by(
             "-start",
             "-end",
-            "delivery_point__station_name",
             "delivery_point__provider_assigned_id",
         )
 
+    def get_consents_stations_grouped(
+        self, consents: Optional[QuerySet["Consent"]] = None
+    ) -> list[Consent] | None:
+        """Get all consents for this entity with grouped stations."""
+        structured_consents: list = []
+
+        if not consents:
+            consents = self.get_consents()
+
+        for consent in consents:
+            stations_grouped = defaultdict(list)
+            for station in consent.delivery_point.stations.all().order_by(
```

Comment:
Order_by should be removed. It generates a large number of queries.

This is done in [next PR](https://github.com/MTES-MCT/qualicharge/pull/496)